### PR TITLE
ddl: fix panic in TestParallelDDL

### LINKED ISSUE #15597: ddl_worker_test.go:testDDLSuite.TestParallelDDL failed
ddl_worker_test.go:testDDLSuite.TestParallelDDL

Latest failed builds: 
https://internal.pingcap.net/idc-jenkins/job/tidb_ghpr_unit_test/28573/display/redirect

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #15597 

Problem Summary:

1. `MockContext` failed interface type assertion, causing a panic. Proof see https://github.com/pingcap/tidb/issues/15597#issuecomment-623045298

2. try to reduce alloc slightly, BTW

### What is changed and how it works?

1. check type assertion and return `statistics.PseudoRowCount` on failure

2. change 'make new slice, append old elements, append new element' to 'append new element to old slice'

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
- No Release note

### CODE REVIEW

**Comment by lance6716** [comment]
PTAL @SunRunAway @wjhuang2016

**Comment by lance6716** [comment]
Almost one-line PR, PTAL @SunRunAway @wjhuang2016

File: ddl/reorg.go [hunk 1]
```diff
@@ -193,8 +193,14 @@ func getTableTotalCount(w *worker, tblInfo *model.TableInfo) int64 {
 	}
 	defer w.sessPool.put(ctx)
 
+	executor, ok := ctx.(sqlexec.RestrictedSQLExecutor)
+	// fix issue 15597
```
Comment (wjhuang2016) [inline ddl/reorg.go:197]: L197 is redundant, please remove it.

Comment (lance6716) [reply ddl/reorg.go]: > L197 is redundant, please remove it.

fixed

File: ddl/reorg.go [hunk 2]
```diff
@@ -193,8 +193,13 @@ func getTableTotalCount(w *worker, tblInfo *model.TableInfo) int64 {
 	}
 	defer w.sessPool.put(ctx)
 
+	executor, ok := ctx.(sqlexec.RestrictedSQLExecutor)
+	// return PseudoRowCount to avoid panic for `ctx` which doesn't implement RestrictedSQLExecutor
```
Comment (zimulala) [inline ddl/reorg.go:197]: I think we can explain that the `ctx` used in the test did not implement this function like `We use `mock.Context` in the test, which does not implement RestrictedSQLExecutor.`

Comment (lance6716) [reply ddl/reorg.go]: done

**Comment by lance6716** [comment]
@bb7133 Hi it's LGT3 in fact

**Comment by zimulala** [comment]
@lance6716 
After CI passes, sre-bot will merge it. Thank you for your PR.

**Pull request merged**